src/test/java/org/folio/fqm/service/FqlToSqlConverterServiceTest.java

```diff
@@ -288,6 +296,40 @@ static List<Arguments> jooqConditionsSource() {
         """
           {"fieldWithFilterValueGetter": {"$eq": "Test value"}}""",
         field("thisIsAFilterValueGetter").eq("Test value".toLowerCase())
+      ),
+
+      Arguments.of(
+        "condition on a field with a filter value getter and a value function",
+        """
+         {
+            "$and": [
+              {"fieldWithFilterValueGetterAndValueFunction": {"$eq": "Test value"}},
+              {"fieldWithAValueFunction": {"$eq": "Test value2"}}
+            ]
+          }""",
+        field("thisIsAFilterValueGetter").eq(field("lower(:value)", param("value", "Test value".toLowerCase())))
+          .and(field("fieldWithAValueFunction").equalIgnoreCase(field("upper(:value)", String.class, param("value", "Test value2"))))
+      ),
+
+      Arguments.of(
+        "in operator on a field with a valueFunction",
+        """
+          {"fieldWithAValueFunction": {"$in": ["value1", 2, true]}}""",
+        or(
+          field("fieldWithAValueFunction").equalIgnoreCase(field("upper(:value)", String.class, param("value", "value1"))),
+          field("fieldWithAValueFunction").eq(field("upper(:value)", String.class, param("value", 2))),
+          field("fieldWithAValueFunction").eq(field("upper(:value)", String.class, param("value", true))))
```

Comment:
Kind of annoying, but you have to do the named parameter binding in the expected results, otherwise the objects aren't considered equivalent